### PR TITLE
This is a preview SDK for the new features introduced in 2018-05-01-preview API version

### PR DESCRIPTION
**Pull request by kalyanaj**
## Description

This is a preview SDK for the new features introduced in 2018-05-01-preview API version, and is expected to be available along with the stable SDK 1.3.0 (already on Nuget). Added corresponding tests, bumped up the version from 1.3.0 to 1.4.0-preview.  Pull Request for Swagger: https://github.com/Azure/azure-rest-api-specs/pull/2796

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]
1. Please create a `generate.ps1` similar to [this](https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/AzureStack/ComputeAdmin/Compute.Admin/generate.ps1) and regenerate the code. Commit any changes to the .txt file
2. Please run `msbuild build.proj /t:Build /p:Scope=SDKs\EventGrid` and commit changes to the .props file

File: src/SDKs/EventGrid/management/EventGrid.Tests/Tests/ScenarioTests.TopicTypeTests.cs [hunk 1]
```diff
@@ -13,32 +13,33 @@ namespace EventGrid.Tests.ScenarioTests
...
+        // [Fact]
+        //public void TopicTypeTests()
+        //{
+        //    const string StorageTopicType = "Microsoft.Storage.StorageAccounts";
+        //    const string EventHubsTopicType = "Microsoft.EventHub.Namespaces";
 
-            using (MockContext context = MockContext.Start(this.GetType().FullName))
-            {
-                InitializeClients(context);
+        //    using (MockContext context = MockContext.Start(this.GetType().FullName))
```
[the remark this answers is not in the document]
Comment (kalyanaj) [reply src/SDKs/EventGrid/management/EventGrid.Tests/Tests/ScenarioTests.TopicTypeTests.cs]: Sure, will update this to skip the test instead.

File: src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj [hunk 2]
```diff
@@ -6,10 +6,10 @@
 	<PropertyGroup>
 		<PackageId>Microsoft.Azure.Management.EventGrid</PackageId>
 		<Description>Provides developers with a library to create and manage all Azure EventGrid resources. </Description>
-		<Version>1.3.0</Version>
+		<Version>1.4.0-preview</Version>
```
[the remark this answers is not in the document]
Comment (kalyanaj) [reply src/SDKs/EventGrid/management/Management.EventGrid/Microsoft.Azure.Management.EventGrid.csproj]: Sure, will change this to 2.0.0-preview.

**Comment by kalyanaj** [comment]
@dsgouda, regarding the first comment "Please create a generate.ps1 and regenerate the code", I want to double-check the right steps with you before I do that. 

1. We have a generate.cmd already. 
2. To generate using this, I will have to first update the autorest configuration in the azure-rest-api-specs repo (README.md) to introduce a new tag for this API-version and then change the default tag from 2018-01-01 to 2018-05-01-preview, correct?

If this is the right approach, I will send out a PR for that autorest configuration change in the azure-rest-api-specs repo.

**Comment by kalyanaj** [comment]
@dsgouda, I have made all the changes except for the props file. Even though I ran "msbuild build.proj /t:Build /p:Scope=SDKs\EventGrid", the \src\SDKs\EventGrid\AzSdk.RP.props file didn't get updated, and that's likely because we have both dataplane and management SDKs, and I remember @shahabhijeet mentioning earlier that some work needed to be done to enable it to be maintained for both data plane and management plane. So, until that's done, I am guessing it is okay.

**Comment by dsgouda** [comment]
@kalyanaj thanks for the feedback

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**